Display the compatibility matrix

0x009922 opened the pull request:
## Linked issue

Closes #355 

## TODO

- [x] Specify the actual URL of the service in `.env`
  - [x] Specified the URL of the test deployment of the service
- [x] Display the matrix in an appropriate place in the docs
- [x] Describe new ENV var in the README

6r1d reviewed (approved):
This matrix is very useful, thanks!

0x009922 commented:
- Updated instructions in README
- Removed "Language-specific Guides" page, created a nested "Compatibility Matrix" section instead
- Updated the look and text in compat matrix section

outoftardis commented on src/guide/compatibility-matrix.md:
```diff
@@ -0,0 +1,18 @@
+
+<CompatibilityMatrixTable />
+
+_Note: The data for this matrix is retrieved dynamically from our [backend service](https://github.com/soramitsu/iroha2-docs-compat-matrix-service), balancing the latest information with a swift response for documentation readers._
```
```suggestion
::: info
The data for this matrix is retrieved dynamically from our [backend service](https://github.com/soramitsu/iroha2-docs-compat-matrix-service), balancing the latest information with a swift response for documentation readers.
:::
```
